exchange/exchange.go

```diff
@@ -99,6 +99,16 @@ func (big *bidIDGenerator) New() (string, error) {
 	return rawUuid.String(), err
 }
 
+type BooleanGenerator interface {
```

Comment:
IMHO - The type name is too generic to understand it's intent. Consider a more specific name such as `Chance`, `Chancer`, or `CoinFlipper`. I also recommend to start with a lower case letter since it's not used my other packages there is no need to export it.

Reply:
Renamed to `deduplicateBidBooleanGenerator`, hope it makes sense

Reply:
That works. Another option might be `deduplicateChanceGenerator`, but now I'm splitting hairs.

Reply:
Ok, renamed